get.sh

```diff
@@ -133,6 +153,66 @@ getTestKitGenAndFunctionalTestMaterial()
 		cd $TESTDIR
 	fi
 
+	if [ "$VENDOR_REPOS" != "" ]; then
+		declare -a vendor_repos_array
+		declare -a vendor_branches_array
+		declare -a vendor_shas_array
+		declare -a vendor_dirs_array
+
+		# convert VENDOR_REPOS to array
+		vendor_repos_array=(`echo $VENDOR_REPOS | sed 's/,/\n/g'`)
+
+		if [ "$VENDOR_BRANCHES" != "" ]; then
+			# convert VENDOR_BRANCHES to array
+			vendor_branches_array=(`echo $VENDOR_BRANCHES | sed 's/,/\n/g'`)
+		fi
+	
+		if [ "$VENDOR_SHAS" != "" ]; then
+			#convert VENDOR_SHAS to array
+			vendor_shas_array=(`echo $VENDOR_SHAS | sed 's/,/\n/g'`)
+		fi
+	
+		if [ "$VENDOR_DIRS" != "" ]; then
+			#convert VENDOR_DIRS to array
+			vendor_dirs_array=(`echo $VENDOR_DIRS | sed 's/,/\n/g'`)
+		fi
+
+		for i in "${!vendor_repos_array[@]}"; do
+			# clone vendor source
+			repoURL=${vendor_repos_array[$i]}
+			branch=${vendor_branches_array[$i]}
+			sha=${vendor_shas_array[$i]}
+			dir=${vendor_dirs_array[$i]}
+			dest="vendor_${i}"
+
+			branchOption=""
+			if [ "$branch" != "" ]; then
+				branchOption="-b $branch"
+			fi
+
+			echo "git clone ${branchOption} $repoURL $dest"
+			git clone -q --depth 1 $branchOption $repoURL $dest
+
+			if [ "$sha" != "" ]; then
+				cd $dest
+				echo "update to $sha"
+				git checkout $sha
+				cd $TESTDIR
+			fi
+
+			# move resources
+			if [[ "$dir" != "" ]] && [[ -d $dest/$dir ]]; then
+				echo "Stage $dest/$dir to openj9/test/functional/"
+				cp -r $dest/$dir openj9/test/functional/
```

Comment:
That won't work. The copy has to be to `openj9/test` not to `TESTDIR`

Reply:
Agree with Lan, we need a solution that will allow us to pull in test material that is not just functional tests.  

However, if this is strictly to overlay more functional tests, is it something that can be done within the functional/build.xml script (@llxia ?)

Reply:
@ShelleyLambert the solution I outlined above does allow to pull any material that is cloned.

Reply:
Yes, but then it appears that whatever material you pick up, you copy it into a functional directory.  

In addition to "functional", we have "system", "jck", "perf", "external", "openjdk" and potentially soon "security" and "fuzz" categories, that we may wish to overlay more test material into, depending on what additional material we pick up and use.

Reply:
@ShelleyLambert  - currently this  pipeline handle only functional because only openj9/functional is staged. Any other directories are removed.

In what I propose, the vendor repo is cloned into a `TESTDIR/vendor` directory. If `dir` is set (dir can be anything we want functional or jdk, etc ) then we copy only `dir` to TESTDIR.
If dir is not set then copy the entire vendor repo content to TESTDIR.

After we clone the vendor repo on the file system we have:
```
TESTDIR
    vendor
        functional
        system
        foo
            bar
    openj9
        test
            functional
            TestConfig
            Utils
 ```            
Let's say I want to stage only foo, thus dir = foo. Will  copy recursively foo to TESTDIR (one directory up) and clean up (rm -fr vendor), thus TESTDIR becomes:
```
TESTDIR
     foo
        bar
     openj9
     ....
```
Then would stage the `openj9/test/TestConfig, openj9/test/Utils, openj9/test/functional` to TESTDIR. Thus TESTDIR becomes:
```
TESTDIR
     foo
        bar
     functional
     TestConfig
     Utils
```

Reply:
@llxia - I have moved the `if [ "$VENDOR_REPOS" != "" ] ... fi ` statement at the end of the function (after the openj9 clone and stage).

Reply:
@llxia ready for your review, thank you!

Reply:
Thanks for the more detailed explanation @vsebe, I will leave it to @llxia for any final/further review comments/questions.  :)